thirdeye/thirdeye-frontend/app/pods/self-serve/import-metric/controller.js

```diff
@@ -1,15 +1,229 @@
+/**
+ * Handles metrics import from inGrahps dashboards
+ * @module self-serve/create/import-metric
+ * @exports import-metric
+ */
 import Ember from 'ember';
+import { checkStatus } from 'thirdeye-frontend/helpers/utils';
 
 export default Ember.Controller.extend({
   /**
-   * Alerts Search Mode
+   * Import Defaults
    */
-  searchModes: ['function-name', 'alert-group'],
+  isImportSuccess: false,
+  isImportError: false,
+  isSubmitDisabled: true,
+  isMetricOnboarded: false,
+  isExistingDashFieldDisabled: false,
+  isCustomDashFieldDisabled: false,
 
+  /**
+   * Enables the submit button when all required fields are filled
+   * @method isSubmitDisabled
+   * @return {Boolean} isDisabled
+   */
+  isSubmitDisabled: Ember.computed(
+    'importExistingDashboardName',
+    'importCustomNewDataset',
+    'importCustomNewRrd',
+    function() {
+      let isDisabled = true;
+      const existingNameField = this.get('importExistingDashboardName');
+      const newNameField = this.get('importCustomNewDataset');
+      const newRrdField = this.get('importCustomNewRrd');
+      // If existing dashboard field is filled, release submit button.
+      if (Ember.isPresent(existingNameField)) {
+        isDisabled = false;
+      }
+      // If any of the 'import custom' fields are filled, assume user will go the RRD import route. Disable submit.
+      if (Ember.isPresent(newNameField) || Ember.isPresent(newRrdField)) {
+        isDisabled = true;
+        // Enable submit if all required RRD fields are present
+        if (Ember.isPresent(newNameField) && Ember.isPresent(newRrdField)) {
+          isDisabled = false;
+        }
+      }
+      return isDisabled;
+    }
+  ),
+
+  /**
+   * Determines whether or not the existing dashboard import mode is active or not.
+   * @method isExistingDashboardNameFieldDisabled
+   * @return {Boolean} isExistingDashFieldDisabled
+   */
+  isExistingDashFieldDisabled: Ember.computed(
+    'importExistingDashboardName',
```

Comment:
Yes, still needed - this is actually the value of the first field, "Import Existing Dashboard"